core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -175,6 +175,13 @@ class Partition(val topic: String,
       // add replicas that are new
       allReplicas.foreach(replica => getOrCreateReplica(replica))
       val newInSyncReplicas = partitionStateInfo.isr.asScala.map(r => getOrCreateReplica(r)).toSet
+
+      // If the ISR size received from a LeaderAndIsrRequest is smaller than the existing ISR size,
+      // mark the isrShrinkRate meter. For example: This can happen if one of the followers is shutdown.
+      // Note that this metric is not triggered on follower -> leader transition
+      // because the inSyncReplicas.size is always zero on a follower
+      if (newInSyncReplicas.size < inSyncReplicas.size) replicaManager.isrShrinkRate.mark()
```

Comment:
Hmm, if the replica is currently a follower, inSyncReplicas will be set to empty as the part of makeFollower(). Then, it seems that newInSyncReplicas.size will always to larger than inSyncReplicas.size?